core/braintree/src/braintree_address.erl

```diff
@@ -92,6 +92,7 @@ update(#bt_address{id=AddressId
 -spec delete(bt_address()) -> bt_address().
 -spec delete(ne_binary() | nonempty_string(), ne_binary() | nonempty_string()) ->  bt_address().
 
+delete(#bt_address{id='undefined'}) -> #bt_address{};
```

Comment:
Could you filter out these beforehand? I would very much prefer we don't add cases like these & just be more careful.